### PR TITLE
New serviceNow template

### PR DESCRIPTION
**Pull request by lrochette**
First draft
Implementation of 3 operations:

1. Create New CR
2. Modify existing CR
3. Close CR

Not yet implemented: resume/kill of workflow (as done in Codefresh Classic). Will come in 2.0.0

### CODE REVIEW

**Comment by lrochette** [comment]
image is incorrect. Pointing to my own repo.
let me fix that

File: workflows/servicenow/versions/1.0.0/workflowTemplate.yaml [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+  name: argo-hub.servicenow.1.0.0
+  annotations:
+    argo-hub/version: '1.0.0'
+    argo-hub/categories: 'utilities'
+    argo-hub/description: 'Execute operations against a Service Now instance'
+    argo-hub/license: 'MIT'
+    argo-hub/owner_name: 'Laurent Rochette'
+    argo-hub/owner_email: 'laurent.rochette@codefresh.io'
+    argo-hub/owner_url: "https://github.com/settings/profile"
+    icon-type: svg
```
Comment (eti-codefresh) [inline workflows/servicenow/versions/1.0.0/workflowTemplate.yaml:13]: need to change to - argo-hub/icon-type

File: workflows/servicenow/versions/1.0.0/workflowTemplate.yaml [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    icon-url: "https://cdn.jsdelivr.net/gh/codefresh-io/argo-hub@main/workflows/servicenow/assets/icon.svg"
```
Comment (eti-codefresh) [inline workflows/servicenow/versions/1.0.0/workflowTemplate.yaml:14]: need to change to - argo-hub/icon_url

File: workflows/servicenow/versions/1.0.0/workflowTemplate.yaml [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+    icon-background: "#81D4FA"
```
Comment (eti-codefresh) [inline workflows/servicenow/versions/1.0.0/workflowTemplate.yaml:15]: need to change to - argo-hub/icon-background

File: workflows/servicenow/versions/1.0.0/workflowTemplate.yaml [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+
+spec:
+  templates:
+    - name: createcr
+      metadata:
+        annotations:
+          icon-type: svg
+          icon-url: "https://cdn.jsdelivr.net/gh/codefresh-io/argo-hub@main/workflows/servicenow/assets/icon.svg"
+          icon-background: "#81D4FA"
```
Comment (eti-codefresh) [inline workflows/servicenow/versions/1.0.0/workflowTemplate.yaml:24]: also all of this argo-hub/

File: workflows/servicenow/versions/1.0.0/workflowTemplate.yaml [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+            value: '{{ inputs.parameters.CR_DATA }}'
+          - name: LOGLEVEL
+            value: '{{ inputs.parameters.LOGLEVEL }}'
+
+    - name: updatecr
+      metadata:
+        annotations:
+          icon-type: svg
+          icon-url: "https://cdn.jsdelivr.net/gh/codefresh-io/argo-hub@main/workflows/servicenow/assets/icon.svg"
+          icon-background: "#81D4FA"
```
Comment (eti-codefresh) [inline workflows/servicenow/versions/1.0.0/workflowTemplate.yaml:70]: and here argo-hub/

File: workflows/servicenow/versions/1.0.0/workflowTemplate.yaml [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+            value: '{{ inputs.parameters.CR_SYSID }}'
+          - name: LOGLEVEL
+            value: '{{ inputs.parameters.LOGLEVEL }}'
+
+    - name: closecr
+      metadata:
+        annotations:
+          icon-type: svg
+          icon-url: "https://cdn.jsdelivr.net/gh/codefresh-io/argo-hub@main/workflows/servicenow/assets/icon.svg"
+          icon-background: "#81D4FA"
```
Comment (eti-codefresh) [inline workflows/servicenow/versions/1.0.0/workflowTemplate.yaml:116]: also here

File: workflows/servicenow/versions/1.0.0/README.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+## Requirements
+1. create a secret containing 2 pieces of informations to connect to your
+ServiceNow instance. See [example](../../assets/sn_auth.yaml)
+  - username
+  - password
+
+## Templates
+1. [createcr](./docs/createcr.md)
+1. [updatecr](./docs/updatecr.md)
+1. [closecr](./docs/closecr.md)
```
Comment (eti-codefresh) [inline workflows/servicenow/versions/1.0.0/README.md:16]: you need to put the actual url- for example
 https://github.com/codefresh-io/argo-hub/blob/main/workflows/servicenow/versions/1.0.0/docs/closecr.md

File: workflows/servicenow/versions/1.0.0/README.md [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+
+## Security
+
+Minimal required permissions
+
+[Full rbac permissions list](./rbac.yaml)
```
Comment (eti-codefresh) [inline workflows/servicenow/versions/1.0.0/README.md:23]: also here

File: workflows/servicenow/versions/1.0.0/README.md (hunk 7 above)
Comment (lrochette) [reply workflows/servicenow/versions/1.0.0/README.md:16]: No, this is a relative path. It works fine

File: workflows/servicenow/versions/1.0.0/README.md (hunk 8 above)
Comment (lrochette) [reply workflows/servicenow/versions/1.0.0/README.md:23]: same, easier to maintain relative paths that hard-coded ones

**Review by eti-codefresh — APPROVED** [review]

**Comment by lrochette** [comment]
Will try to do a PR from fork to local branch
Then PR from that branch to main
Therefore closing that PR

**Pull request closed without merging**